src/persist-client/src/internal/state.proto

```diff
@@ -76,14 +76,19 @@ message ProtoIdSpineBatch {
     ProtoSpineBatch batch = 2;
 }
 
-message ProtoFuelingMerge {
```

Comment:
Another upshot is a forwards-compatibility risk - if we enable the flags prematurely, we will serialize `ProtoMerge`s that old code won't know how to read.

This is relatively easy to deal with - just don't enable the flag until this code has been fully rolled out - but it takes this proto change for me from "obviously worth it" to "reasonably worth it". Feel free to push back if you feel differently!

Reply:
Bypassing the check sounds good to me!

> we will serialize `ProtoMerge`s that old code won't know how to read.

If the old code is deserializing the encoded bytes of a `ProtoMerge` as a `ProtoFuelingMerge` then I think we should be okay? The old code will just ignore the `active_compaction` field

Reply:
The issue is that we end up with `ThinMerge`s that don't correspond to any fueling merge (because they're actually in-progress compactions) but `unflatten` asserts that there aren't any merges left when it's done deserializing things. (This isn't just me being smart and thinking ahead; this was a bug I ran into in nightlies.)